frontend/src/reducers/discover.js

```diff
@@ -0,0 +1,27 @@
+import merge from 'lodash/object/merge';
+import { DISCOVER_SUCCESS } from '../constants/discover';
+
+export default function discover(state={}, action={}) {
+  switch (action.type) {
+    case DISCOVER_SUCCESS:
+    const discover = action.json;
+    if (state.show) {
+      if (state.show === discover.show && state.sort === discover.sort) {
+        if (state.collectives) {
+          for (let i = state.collectives.length - 1; i >= 0; i--) {
+            discover.collectives.unshift(state.collectives[i]);
+          }
+          return merge({}, state, discover);
+        } else {
+          return merge(state, discover);
```

Comment:
Is there a reason why I should always merge into a new object that is, return a new object every time?

As explained above, this is the condition where the page has just loaded and `state.collectives` is undefined.
There I merged the new results directly into the state object that is returned, thats all. 

I don't see why we always create a new object, other than, thats just how its always been?